native/test/src/tests/room/policy_rule_room.rs

```diff
@@ -0,0 +1,106 @@
+use acter::api::TimelineItem;
+use anyhow::Result;
+use core::time::Duration;
+use futures::{pin_mut, stream::StreamExt, FutureExt};
+use tokio::time::sleep;
+use tokio_retry::{
+    strategy::{jitter, FibonacciBackoff},
+    Retry,
+};
+
+use crate::utils::random_user_with_random_convo;
+
+#[tokio::test]
+async fn test_policy_rule_room() -> Result<()> {
+    let _ = env_logger::try_init();
+
+    let (mut user, room_id) = random_user_with_random_convo("policy_rule_room").await?;
+    let state_sync = user.start_sync();
+    state_sync.await_has_synced_history().await?;
+
+    // wait for sync to catch up
+    let retry_strategy = FibonacciBackoff::from_millis(100).map(jitter).take(10);
+    let fetcher_client = user.clone();
+    let target_id = room_id.clone();
+    Retry::spawn(retry_strategy, move || {
+        let client = fetcher_client.clone();
+        let room_id = target_id.clone();
+        async move { client.convo(room_id.to_string()).await }
+    })
+    .await?;
+
+    let convo = user.convo(room_id.to_string()).await?;
+    let timeline = convo.timeline_stream();
+    let stream = timeline.messages_stream();
+    pin_mut!(stream);
+
+    let policy_event_id = convo
+        .set_policy_rule_room(
+            "#*:example.org".to_owned(),
+            "undesirable content".to_owned(),
+        )
+        .await?;
+
+    // room state event may reach via pushback action or reset action
+    let mut i = 30;
+    let mut found_event_id = None;
+    while i > 0 {
+        if let Some(diff) = stream.next().now_or_never().flatten() {
+            match diff.action().as_str() {
+                "PushBack" | "Set" => {
+                    let value = diff
+                        .value()
+                        .expect("diff pushback action should have valid value");
+                    if let Some(event_id) = match_msg(&value, "Set", "#*:example.org") {
+                        found_event_id = Some(event_id);
+                    }
+                }
+                "Reset" => {
+                    let values = diff
+                        .values()
+                        .expect("diff reset action should have valid values");
+                    for value in values.iter() {
+                        if let Some(event_id) = match_msg(value, "Set", "#*:example.org") {
+                            found_event_id = Some(event_id);
+                            break;
+                        }
+                    }
+                }
+                _ => {}
+            }
+            // yay
+            if found_event_id.is_some() {
+                break;
+            }
+        }
+        i -= 1;
+        sleep(Duration::from_secs(1)).await;
+    }
+    assert_eq!(
+        found_event_id,
+        Some(policy_event_id.to_string()),
+        "Even after 30 seconds, policy rule room not received",
+    );
```

Comment:
let's add more checking of the actual values of the API coming through. too little of the policy_rule_room model is covered. add tests for these methods to confirm their values, please

Reply:
Fixed by https://github.com/acterglobal/a3/pull/2839/commits/c4c1f754c5947297bdeaedd8491ba01dd0d879c5